parquet-hadoop/src/main/java/parquet/hadoop/MemoryManager.java

```diff
@@ -155,4 +160,31 @@ long getTotalMemoryPool() {
   float getMemoryPoolRatio() {
     return memoryPoolRatio;
   }
+
+  /**
+   * Register callback and deduplicate it if any.
+   * @param callBackName the name of callback. It should be identical.
+   * @param callBack the callback passed in from upper layer, such as Hive.
+   * @return
+   */
+  public boolean registerScaleCallBack(String callBackName, Runnable callBack) {
```

Comment:
how about making this function void, using preconditions to assert that callBackName and callBack are not null, and throwing when two callbacks are assigned the same name?